flutter/shell/platform/tizen/flutter_tizen_view.cc

```diff
@@ -94,13 +90,13 @@ void FlutterTizenView::CreateRenderSurface(
 
   if (engine_ && engine_->renderer()) {
     TizenGeometry geometry = tizen_view_->GetGeometry();
-    if (tizen_view_->GetType() == TizenViewType::kWindow) {
-      auto* window = reinterpret_cast<TizenWindow*>(tizen_view_.get());
+    if (dynamic_cast<TizenWindow*>(tizen_view_.get())) {
```

Comment:
`typeid` should be better in this specific case.

https://en.wikipedia.org/wiki/Run-time_type_information

> The use of `typeid`, in a non-polymorphic context, is often preferred over `dynamic_cast<class_type>` in situations where just the class information is needed, because `typeid` is always a [constant-time](https://en.wikipedia.org/wiki/Constant_time) procedure

Reply:
> typeid should be better in this specific case.

Oh... The type of `tizen_view_` is `std::unique_ptr<TizenViewBase>`.
I think, `typeid()` didn't make sense in this case because it returns `std::unique_ptr<TizenViewBase>`.
The reason `dynamic_cast` is used here is that it returns null if `tizen_view_` cannot be casted to `TizenWindow*`.

Reply:
Hmm? Does `typeid(tizen_view_.get())` return the typeid of `TizenViewBase`?

Reply:
I was test this
```cpp
FT_LOG(Error) << "CJS typeid(renderer_.get()) : "
                << typeid(renderer_.get()).name();
FT_LOG(Error) << "CJS typeid(TizenRendererEvasGL).name() : "
                << typeid(TizenRendererEvasGL).name();

auto& casted_renderer = *renderer_.get();
FT_LOG(Error) << "CJS typeid(casted_renderer) : "
              << typeid(casted_renderer).name();
FT_LOG(Error)
    << "CJS (typeid(casted_renderer) == typeid(TizenRendererEvasGL)) : "
    << (typeid(casted_renderer) == typeid(TizenRendererEvasGL));
```
```kernel
[E] flutter_tizen_engine.cc: RunEngine(163) > CJS typeid(renderer_.get()) : PN7flutter13TizenRendererE
[E] flutter_tizen_engine.cc: RunEngine(165) > CJS typeid(TizenRendererEvasGL).name() : N7flutter19TizenRendererEvasGLE
[E] flutter_tizen_engine.cc: RunEngine(169) > CJS typeid(casted_renderer) : N7flutter19TizenRendererEvasGLE
[E] flutter_tizen_engine.cc: RunEngine(171) > CJS (typeid(casted_renderer) == typeid(TizenRendererEvasGL)) : 1
```

So we can use it like this.
```cpp
auto& casted_renderer = *renderer_.get();
if (IsHeaded() && (typeid(casted_renderer) == typeid(TizenRendererEvasGL))) {
```
but, I think it's better now than creating a new local variable.

+)
`typeid(*renderer_.get())` is make  `-Werror,-Wpotentially-evaluated-expression`